src/components/Trade/hooks/useSwapper/useSwapper.tsx

```diff
@@ -151,6 +166,76 @@ export const useSwapper = () => {
     ],
   )
 
+  const checkApprovalNeeded = useCallback(async () => {
+    const adapterManager = getChainAdapterManager()
+    const adapter = adapterManager.get(sellAsset.chainId)
+
+    if (!adapter) throw Error(`no chain adapter found for chain Id: ${sellAsset.chainId}`)
+    if (!wallet) throw new Error('no wallet available')
+    if (!activeQuote) throw new Error('no activeQuote available')
+
+    // No approval needed for selling a fee asset
+    if (sellAsset.assetId === adapter.getFeeAssetId()) {
+      return false
+    }
+
+    const ownerAddress = await adapter.getAddress({
+      wallet,
+      accountNumber: activeQuote.accountNumber,
+    })
+
+    const { assetReference: sellAssetErc20Address } = fromAssetId(sellAsset.assetId)
+    const web3 = getWeb3InstanceByChainId(sellAsset.chainId)
+
+    const allowanceResult = await getERC20Allowance({
+      web3,
+      erc20AllowanceAbi,
+      sellAssetErc20Address,
+      spenderAddress: activeQuote.allowanceContract,
+      ownerAddress,
+    })
+    const allowanceOnChain = bnOrZero(allowanceResult)
```

Comment:
Should be able to use `bn` according to types, `bnOrZero` makes it look like the intent is to potentially do a `getOrElse(0)` on the allowance.
With that being said, while the types tell us this is a number, a JSON-RPC call can fail for any reason and we may want to try/catch on the `getERC20Allowance` call as a paranoia check, which we never did

Reply:
Also not introduced by this PR but whilst we're in the neighborhood could use the base unit vernacular to make it clear this is a non-massaged wei amount, since we're doing a lt check against `sellAmountBeforeFeesCryptoBaseUnit` on l.199